From 2b906cc94fb2a484cd0a8237497b09f6bbff0db5 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 22 Oct 2026 09:36:28 +0800
Subject: [PATCH 2/2] AOSCOS: make-ca: drop -t option from mktemp

According to Beyond Linux From Scratch version 13.1, Chapter 4 on
"make-ca-1.16.1":

  At first, remove the -t option from the mktemp commands in the script.
  This option is deprecated and it can cause unwanted effects if TMPDIR
  is set in the environment.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 make-ca | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/make-ca b/make-ca
index 437b1b9..d8c6da6 100644
--- a/make-ca
+++ b/make-ca
@@ -748,7 +748,7 @@ CERTBEGINLIST=`grep -n "^# Certificate" "${WORKDIR}/certdata.txt" | \
 
 # Dump individual certs to temp file
 for certbegin in ${CERTBEGINLIST}; do
-  tmpcrtbeg=$(mktemp -p ${TEMPDIR}/certs -t make-ca.XXXXXXXX.tmp)
+  tmpcrtbeg=$(mktemp -p ${TEMPDIR}/certs make-ca.XXXXXXXX.tmp)
   awk "NR==$certbegin,/^CKA_TRUST_STEP_UP_APPROVED/" "${WORKDIR}/certdata.txt" \
       > $tmpcrtbeg
 done
@@ -916,7 +916,7 @@ if test -d "${LOCALDIR}"; then
     write_anchor
 
     # Generate working copy
-    tempcert=$(mktemp -p ${TEMPDIR}/certs -t make-ca.XXXXXXXX.pem)
+    tempcert=$(mktemp -p ${TEMPDIR}/certs make-ca.XXXXXXXX.pem)
     "${OPENSSL}" x509 -in "${cert}" -text -fingerprint > ${tempcert}
 
     # Add to Shared NSS DB
-- 
2.56.0

